lib/src/git.rs

```diff
@@ -1319,16 +1322,45 @@ fn push_refs(
             GitPushError::InternalGitError(err)
         }
     })?;
-    let mut remaining_remote_refs: HashSet<_> = qualified_remote_refs.iter().copied().collect();
+    let mut remaining_remote_refs_expected_locations =
```

Comment:
nit: maybe better to leave `remaining_remote_refs` as `HashSet<_>`? We have to create an owned set or map anyway to get around mutable borrows.